storage.go

```diff
@@ -259,7 +256,9 @@ func NewStorageFromHTTP(u *url.URL, options HTTPClientStorageOptions) (Storage,
 		if err != nil {
 			return fmt.Errorf("failed to decode JWK Set response: %w", err)
 		}
-		store.KeyDeleteAll() // Clear local cache in case of key revocation.
```

Comment:
There is a minuscule chance that a read could take place during the execution of the below loop. That read would be missing one or more keys.

An unlikely race condition. Not one that would corrupt memory, due to the locking within the methods.

The worst consequence I can see is a few microseconds (milliseconds?) of rejecting JWK that should be accepted.